pkg/controller/perconaservermongodb/finalizers.go

```diff
@@ -145,13 +160,32 @@ func (r *ReconcilePerconaServerMongoDB) deleteRSPods(ctx context.Context, cr *ap
 			return errors.Wrap(err, "is pod primary")
 		}
 		if !isPrimary {
-			return errWaitingTermination
+			return errWaitingFirstPrimary
 		}
 
 		// If true, we should resize the replset to 0
 		rs.Size = 0
 		return errWaitingTermination
+	case rs.Size:
+		// If statefulset size is equal to size of replset, that means that we just started the execution of finalizer
+		// and firstly we set the first pod as primary.
+		// After that we can start resizing the statefulset.
+		isPrimary, err := r.isPodPrimary(ctx, cr, pods.Items[0], rs)
+		if err != nil {
+			return errors.Wrap(err, "is pod primary")
+		}
```

Comment:
Yes because we can't resize the statefulset just after the `setPrimary` call as it doesn't set primary instantly.

I would remove that extra check in the `setPrimary` func, but if we plan to reuse it, I think we should leave it as it is